cbor/shared/src/main/scala/fs2/data/cbor/high/internal/ValueSerializer.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data
+package cbor
+package high
+package internal
+
+import low.CborItem
+
+import java.lang.{Float => JFloat, Double => JDouble}
+
+import scodec.bits._
+
+private[cbor] object ValueSerializer {
+
+  private val one = BigInt(1)
+
+  def toItems[F[_]]: Pipe[F, CborValue, CborItem] = {
+    def go(chunk: Chunk[CborValue],
+           idx: Int,
+           rest: Stream[F, CborValue],
+           acc: List[CborItem]): Pull[F, CborItem, Unit] =
+      if (idx >= chunk.size) {
+        Pull.output(Chunk.seq(acc.reverse)) >> rest.pull.uncons.flatMap {
+          case Some((hd, tl)) => go(hd, 0, tl, Nil)
+          case None           => Pull.done
+        }
+      } else {
+        chunk(idx) match {
+          case CborValue.Integer(i) =>
+            // if representation is too big, use the tags specified in the RFC
+            if (i >= 0) {
+              val bytes = ByteVector(i.toByteArray)
+              if (bytes.size <= 64)
```

Comment:
64 _bytes_? I guess you have the green RFC tests, but I would have expected bits.

Reply:
I used to encode as `BitVector` and forgot to change this check. Thanks.

Reply:
Make sure to check the whole method, there are some more magic numbers in this method that look suspiciously like bit counts.